### PR TITLE
Update revision 1.0.0 rc.2 [ECR-4274]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

- Anchoring disabled
- ~~`run-dev` command support is broken and disabled~~
- ~~TestKit services instantiation is complete nightmare and would be fixed~~
- Transactions errors records are currently broken, affected tests disabled.

Core revision updated to the most recent master one (pre-1.0.0-rc.3)

---
See: https://jira.bf.local/browse/ECR-4274

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
> TestKit services instantiation is complete nightmare and would be fixed

Why did the core make it a nightmare?

File: exonum-java-binding/core/rust/exonum-java/Cargo.toml [hunk 1]
```diff
@@ -5,15 +5,19 @@ authors = ["Exonum team <contact@exonum.com>"]
...
 # Although its primary goal is to detect _programming_ errors that shall not occur
 # in a correct application early, it is currently required in the application
 # because some public APIs in Java permit passing arbitrary pointers (e.g., Snapshot#newInstance).
 java_bindings = { path = "..", features = ["resource-manager"] }
 log = "0.4"
 serde = { version = "1.0", features = ["serde_derive"] }
+
+[dependencies.tokio]
+version = "0.2.13"
+features = ["blocking", "dns", "io-util", "macros", "rt-threaded", "tcp", "time"]
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/exonum-java/Cargo.toml]: How are these features picked?

File: exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java [hunk 2]
```diff
@@ -589,23 +589,22 @@ private TestKitServiceInstances prepareServicesConfiguration() {
       checkDeployedArtifactsAreUsed();
       TestKitServiceInstances.Builder builder = TestKitServiceInstances.newBuilder();
 
-      // Add specifications of artifacts to deploy.
-      for (Map.Entry<ServiceArtifactId, String> entry : serviceArtifactFilenames.entrySet()) {
-        ServiceArtifactId artifactId = entry.getKey();
+      for (ServiceArtifactId artifactId : services.keySet()) {
+        TestKitProtos.TestKitService.Builder serviceBuilder =
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java]: I'd initialize it below, closer to where it is used.

**Comment by dmitry-timofeev** [comment]
Also, what did again happen to the BTC anchoring?

**Comment by vitvakatu** [comment]
> Also, what did again happen to the BTC anchoring?

Actually nothing, it just wasn't updated to rc.2 when I started this PR, and later I updated to most recent master, so we can't use it again. We will reenable anchoring with rc.3

File: exonum-java-binding/core/rust/exonum-java/Cargo.toml (hunk 1 above)
Comment (vitvakatu) [reply exonum-java-binding/core/rust/exonum-java/Cargo.toml]: It was actually copy-pasted from Core's cryptocurrency example... I'll check if we can get rid of them

**Comment by vitvakatu** [comment]
Ah okay, Travis won't trigger build for this PR. Not a big deal I think, I checked exonum-java locally, it at least builds

**Pull request merged**